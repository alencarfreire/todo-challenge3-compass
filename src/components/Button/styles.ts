import styled from "styled-components/native";
import { theme } from "../../theme/theme";

interface ButtonProps {
  padding?: string;
}

export const Button = styled.TouchableOpacity<ButtonProps>`
  background-color: ${theme.colors.main.purpleDark};
  padding: ${({ padding }: { padding: string }) =>
    padding ? padding : "21px"};
  gap: 8px;
  justify-content: center;
  align-items: center;
  border-radius: 8px;
  flex-direction: row;
`;

export const ButtonText = styled.Text`
  color: ${theme.colors.base.gray100};
  line-height: ${theme.font.lineHeight.button};
  font-size: ${({ variant }: { variant: string }) =>
    variant === "Authentication"
      ? theme.font.size.input
      : theme.font.size.button};
  font-family: ${({ variant }: { variant: string }) =>
    variant === "Authentication"
      ? theme.font.family.input
      : theme.font.family.button};
`;
