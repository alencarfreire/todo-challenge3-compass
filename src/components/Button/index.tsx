import React from "react";
import * as S from "./styles";

type Props = {
  text?: string;
  children?: React.ReactNode;
  onPress?: () => void;
  variant?: "Authentication" | "home";
  padding?: string;
};

export default function Button({
  text,
  children,
  onPress,
  variant,
  padding,
}: Props) {
  return (
    <>
      <S.Button onPress={onPress} padding={padding} testID="button-component">
        {text && <S.ButtonText variant={variant}>{text}</S.ButtonText>}
        {children}
      </S.Button>
    </>
  );
}
