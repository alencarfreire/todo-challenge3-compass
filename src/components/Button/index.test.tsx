import { render, fireEvent } from "@testing-library/react-native";
import { Text } from "react-native";
import Button from "./index";

describe("Component: Button", () => {
  it("should render the Button component", () => {
    const { getByTestId } = render(<Button />);
    expect(getByTestId("button-component")).toBeTruthy();
  });

  it("should render text when the text prop is provided", () => {
    const text = "Clique aqui";
    const { getByText } = render(<Button text={text} />);
    expect(getByText(text)).toBeTruthy();
  });

  it("should render children when passed", () => {
    const children = <Text>Child Element</Text>;
    const { getByText } = render(<Button>{children}</Button>);
    expect(getByText("Child Element")).toBeTruthy();
  });

  it("should trigger onPress when the button is pressed", () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(<Button onPress={onPressMock} />);
    const button = getByTestId("button-component");
    fireEvent.press(button);
    expect(onPressMock).toHaveBeenCalled();
  });

  it("should render with a specific variant style", () => {
    const variant = "Authentication";
    const text = "Submit";
    const { getByText } = render(<Button text={text} variant={variant} />);
    const buttonText = getByText(text);
    expect(buttonText.props.variant).toBe(variant);
  });

  it("should render with custom padding when provided", () => {
    const padding = "16px";
    const { toJSON } = render(<Button padding={padding} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
