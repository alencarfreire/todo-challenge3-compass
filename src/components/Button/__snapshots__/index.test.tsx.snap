// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Button should render with custom padding when provided 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#6F3CC3",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "flexDirection": "row",
      "gap": 8,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
  testID="button-component"
/>
`;
